code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm

```diff
@@ -299,21 +299,149 @@
-	attack_verb_simple = "bounce off of"
-	loot = list(/obj/item/clothing/mask/gas/clown_hat, /obj/effect/gibspawner/xeno/bodypartless, /obj/effect/particle_effect/foam, /obj/item/soap, /obj/effect/gibspawner/generic, /obj/effect/gibspawner/generic/animal, /obj/effect/gibspawner/human/bodypartless, /obj/effect/gibspawner/human)
-	attack_reagent = /datum/reagent/toxin/mindbreaker
-
-/mob/living/simple_animal/hostile/retaliate/clown/mutant/blob/slow
-	speed = 50
-	move_to_delay = 120
+	speed = 1
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	force_threshold = 10 //lots of fat to cushion blows.
+	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 1, CLONE = 2, STAMINA = 0, OXY = 1)
+	attack_verb_continuous = "slams"
+	attack_verb_simple = "slam"
+	loot = list(/obj/effect/gibspawner/xeno/bodypartless, /obj/effect/gibspawner/generic, /obj/effect/gibspawner/generic/animal, /obj/effect/gibspawner/human/bodypartless)
+	deathsound = 'sound/misc/sadtrombone.ogg'
+	food_type = list(/obj/item/food/cheesiehonkers, /obj/item/food/cornchips)
+	tame_chance = 30
+	///This is the list of items we are ready to regurgitate,
+	var/list/prank_pouch = list()
+	///This ability lets you fire a single random item from your pouch.
+	var/obj/effect/proc_holder/regurgitate/my_regurgitate
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/Initialize()
+	. = ..()
+	my_regurgitate = new
+	AddAbility(my_regurgitate)
+	add_cell_sample()
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/attacked_by(obj/item/I, mob/living/user)
+	if(!check_edible(I))
+		return ..()
+	eat_atom(I)
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/AttackingTarget(atom/attacked_target)
+	if(!check_edible(attacked_target))
+		return ..()
+	eat_atom(attacked_target)
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/UnarmedAttack(atom/A)
+	if(!check_edible(A))
+		return ..()
+	eat_atom(A)
+
+///Returns whether or not the supplied movable atom is edible.
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/proc/check_edible(atom/movable/potential_food)
+	if(isliving(potential_food))
+		var/mob/living/living_morsel = potential_food
+		if(living_morsel.mob_size > MOB_SIZE_SMALL)
+			return FALSE
+		else
+			return TRUE
+
+	if(IS_EDIBLE(potential_food))
+		if(prank_pouch.len >= 8)
+			to_chat(src, "<span class='warning'>Your prank pouch is filled to the brim! You don't think you can swallow any more morsels right now.</span>")
+			return FALSE
+		return TRUE
+
+///This proc eats the atom, certain funny items are stored directly in the prank pouch while bananas grant a heal based on their potency and the peels are retained in the pouch.
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/proc/eat_atom(atom/movable/eaten_atom)
+	if(istype(eaten_atom, /obj/item/food/pie/cream) || istype(eaten_atom, /obj/item/food/grown/tomato) || istype(eaten_atom, /obj/item/food/meatclown))
```

Comment:
Would be better cleaner if the allowed objects were a static list.

Reply:
Done